File: java/src/yax/lex/SaxEvent.java
/* Copyright 2009, UCAR/Unidata and OPeNDAP, Inc.
   See the COPYRIGHT file for more information. */

package yax.lex;

import org.xml.sax.Locator;

/**
Define a class to hold information provided to each kind of event
*/
public class SaxEvent
{
    public Type type = null;
    public SaxEventType event = null;
    public String name = null;
    public String fullname = null;
    public String namespace = null;

    public String value = null; // for attributes
    public String text = null;  // for text

    public String publicid = null;
    public String systemid = null;

    public Locator locator = null;

    public SaxEvent(SaxEventType event, Locator locator)
    {
	this.event = event;
	this.locator = locator;
	this.type = event.getType();
    }

    public SaxEvent(SaxEventType event, Locator locator, String name)
    {
        this(event,locator);
	this.name=name;
    }

    public SaxEvent(SaxEventType event, Locator locator,
                    String name, String fullname, String uri)
    {
	this(event,locator,name);
	this.fullname = fullname;
	this.namespace = uri;
    }

    public String toString()
    {
	String text = (type == null ? "undefined" : type.toString());
	return text;
    }

} // class SaxEvent

File: python/yax/saxevent.py
# Copyright 2013, UCAR/Unidata.
# See the COPYRIGHT file for more information.

# Define a class to hold information provided to each kind of event

import saxeventtype

class SaxEvent:

  def __init__(self, event, locator=None, name=None, fullname=None, uri=None,
               text=None, value=None):
    self.event = event
    self.locator = locator
    self.name = name
    self.fullname = fullname
    self.namespace = uri
    self.text = text  # for text
    self.value = value # for attributes
    self.publicid = None
    self.systemid = None
  # end __init__

  def __str__(self) :
    if self.event == None :
      text = "undefined"
    else :
      text = saxeventtype.tostring(self.event)
    return text
  # end __str__

# class SaxEvent

